### PR TITLE
docs: Add Zenodo metadata file

### PR DESCRIPTION
**Pull request by matthewfeickert**
This PR enforces a standard for the Zenodo metadata and makes the resulting DOI entry much nicer &mdash; the authors listed additionally have their ORCID listed on the Zenodo DOI entry. c.f. `pyhf`'s DOI for an example: [DOI](https://doi.org/10.5281/zenodo.1169739)

**Suggested squash and merge commit message**:
```
* Add .zenodo.json to control metadata for releases on Zenodo
* Control .zenodo.json with bump2version
* Add Matthew as author
```

### CODE REVIEW

File: .zenodo.json
Comment (matthewfeickert) [inline .zenodo.json:9]: @dguest Probably want to change this to wherever you're postdocing now. I'm not sure where in Germany you are and the new CERN phonebook style has removed seeing where people are affiliated with.

Comment (dguest) [reply .zenodo.json:9]: Humboldt-Universität zu Berlin, but I can't even type umlauts with the way my keyboard is set up. On other packages I just left out the affiliation and used the orcid. But yeah I should update that.

**Review by dguest — COMMENTED** [review]
I'm not going to merge this unless you add yourself.

**Comment by matthewfeickert** [comment]
> I'm not going to merge this unless you add yourself.

I think that's overly generous of you, but done.

**Review by dguest — APPROVED** [review]
Awesome.

**Comment by dguest** [comment]
Actually, one comment: the old record also lists Lawrence Lee; Alex Armstrong; Giordon Stark. I don't think they count as maintainers so you don't have to put them everywhere, but my inclination is to add people who did _anything_ early on to the "creators" field since that gives them a small chance of some kind of indirect credit via the computer bot web.

I like giving people credit via the indirect computer bot web, even if all they did is make a pull request that's more than almost any software user does.

**Comment by matthewfeickert** [comment]
> I like giving people credit via the indirect computer bot web, even if all they did is make a pull request that's more than almost any software user does.

@dguest Yeah, this is one of the things that gets tricky to some extent. They're always be on the older Zenodo archives (as they're archives) but it gets a bit confusing if you're trying to use a Zenodo citation and there's many people attached to know who to actually cite in addition to the "et al.".

One way that people on Scikit-HEP are starting to do this is with making it explicit what people did in the [Acknowledgements section of the `README`](https://github.com/scikit-hep/awkward-1.0#acknowledgements) and then keeping the dev team as the creators and maintainers on PyPI and Zenodo.

**Pull request merged**
